tests/e2e/helpers/dropPostgresDatabase.ts

```diff
@@ -0,0 +1,12 @@
+import { PrismaClient } from '@prisma/client'
+
+export async function dropPostgresDatabase(prismaAdminClient: PrismaClient, databaseName: string) {
+  try {
+    await prismaAdminClient.$executeRawUnsafe(`DROP DATABASE ${databaseName} WITH (FORCE);`)
+  } catch (error) {
+    const isDatabaseNotFoundErorr = error instanceof Error && error.message.match(/does not exist/)
+    if (!isDatabaseNotFoundErorr) throw error
+  } finally {
+    await prismaAdminClient.$disconnect()
+  }
+}
```

Comment:
How about a `dropDatabase` function that accepts a `datasourceProvider` type which this then switches on. Then it becomes an implementation detail of `testTemplate`.

`testTemplate` was previously generalized over dataSource provider and I _think_ it should stay that way.

So in other words I think `databaseActions` of `testTemplate` should not exist.

Reply:
I'd say that would be more complex to read than splitting in two methods, because the switch cases will increase in time with each db, so it is going agains SOLID principles. I prefer using dependencies which are passed in the method instead of accessing global stuff especially when the method `testTemplate` is being widely used.
Too generic is never a good thing. (it does simplify how you use it, but it hides all the magic code with if/else statements inside on which the dev relies on later, so when a new dev has to debug things it makes things more difficult) - Also another good practice is, making things open for extending but closed for modification.

If you insist to have it like that I'll refactor it.

Reply:
The goal of `testTemplate` is to be a top level declarative function. Tell which template to test under what set of inputs and it figures out the rest.

How it is implemented is another story, and its not worth discussing before aligning on this initial API goal.

Reply:
As per conversation from Slack - Agreed to use the closed approach and define the dropDatabase and initDatabase within the testTemplate method.